modules/spring-config/src/main/java/org/mule/config/spring/SpringRegistryBootstrap.java

```diff
@@ -0,0 +1,489 @@
+import java.util.Map.Entry;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+
+/**
+ * This object will generate bean definitions for entries defined in a file called <code>registry-bootstrap.properties</code>.
+ * This allows modules and transports to make certain objects available by default.  The most common use case is for a
+ * module or transport to load stateless transformers into the registry.
+ * For this file to be located it must be present in the modules META-INF directory under
+ * <pre>META-INF/services/org/mule/config/</pre>
+ * <p/>
+ * The format of this file is a simple key / value pair. i.e.
+ * <pre>
+ * myobject=org.foo.MyObject
+ * </pre>
+ * Will register an instance of MyObject with a key of 'myobject'. If you don't care about the object name and want to
+ * ensure that the object gets a unique name you can use -
+ * <pre>
+ * object.1=org.foo.MyObject
+ * object.2=org.bar.MyObject
+ * </pre>
+ * or
+ * <pre>
+ * myFoo=org.foo.MyObject
+ * myBar=org.bar.MyObject
+ * </pre>
+ * It's also possible to define if the entry must be applied to a domain, an application, or both by using the parameter applyToArtifactType.
+ * <pre>
+ * myFoo=org.foo.MyObject will be applied to any mule application since the parameter applyToArtifactType default value is app
+ * myFoo=org.foo.MyObject;applyToArtifactType=app will be applied to any mule application
+ * myFoo=org.foo.MyObject;applyToArtifactType=domain will be applied to any mule domain
+ * myFoo=org.foo.MyObject;applyToArtifactType=app/domain will be applied to any mule application and any mule domain
+ * </pre>
+ * Loading transformers has a slightly different notation since you can define the 'returnClass' with optional mime type, and 'name'of
+ * the transformer as parameters i.e.
+ * <pre>
+ * transformer.1=org.mule.transport.jms.transformers.JMSMessageToObject,returnClass=byte[]
+ * transformer.2=org.mule.transport.jms.transformers.JMSMessageToObject,returnClass=java.lang.String:text/xml, name=JMSMessageToString
+ * transformer.3=org.mule.transport.jms.transformers.JMSMessageToObject,returnClass=java.util.Hashtable)
+ * </pre>
+ * Note that the key used for transformers must be 'transformer.x' where 'x' is a sequential number.  The transformer name will be
+ * automatically generated as JMSMessageToXXX where XXX is the return class name i.e. JMSMessageToString unless a 'name'
+ * parameter is specified. If no 'returnClass' is specified the default in the transformer will be used.
+ * <p/>
+ * Note that all objects defined have to have a default constructor. They can implement injection interfaces such as
+ * {@link MuleContextAware} and lifecycle interfaces such as {@link Initialisable}.
+ *
+ * @since 3.7.0
+ */
+public class SpringRegistryBootstrap implements MuleContextAware
```

Comment:
Worth adding RegistyBootstrap interface in mule-core registry package?  I would.